server/tabby_server/vision/extraction.py

```diff
@@ -0,0 +1,197 @@
+
+_ANSWER_COUNT = 5
+"""Number of answers to get from ChatGPT."""
+
+_SEPERATOR = "|---|"
+"""Separator for input and output messages."""
+
+_SYSTEM_MESSAGE = f"""\
+You are a model which accepts chunks of text which were recognized by an OCR model. The texts will be from the cover of a physical book. Using your knowledge of natural language and the internet, you will identify (1) the title and (2) the author, given the text.
+
+In the input, you will accept 1 or more lines of text. Conditions:
+- Each line is in the format of "TEXT |---| AREA |---| CENTER_X, CENTER_Y"
+- TEXT is the text recognized by the OCR model.
+- There may be misspellings in the text, for which you must account for.
+- There may be parts of text which do are not part of the title nor the author.
+- AREA is a floating point value representing how much area the text takes up.
+- Texts with larger area tend to be part of the title or author.
+- CENTER_X, CENTER_Y are floating point values which represent the center point of the text's bounding box.
+- Texts which have close centers may be a part of a larger chunk.
+
+You will output {_ANSWER_COUNT} answers on separate lines. Conditions:
+- Each answer is in the format of "TITLE |---| AUTHOR". YOU MUST STRICTLY OBEY THIS FORMAT. Do not number or bulletpoint each answer.
+- Each answer is unique.
+- Every character should be UPPERCASE.
+- The first answer is your most confident answer, while the bottom answer is your least confident answer.
+- More confident answers should include the volume.
+- More confident answers should include the edition.
+"""  # noqa: E501
+
+#
+# Used to be part of system message:
+#
+# Finally, you must output a separate line after your answers a message explaining your choices for your answers. Conditions:  # noqa: E501
+# - This must be in the format of "Explanation: <EXPLANATION MESSAGE>"
+#
+# This was removed as it made ChatGPT take more time to generate a response to
+# this.
+#
+
+
+@dataclass
+class ExtractionOption:
+    """Represents a single option (out of several) which could be correct."""
+
+    title: str
+    """Extracted title."""
+
+    author: str
+    """Extracted author."""
+
+
+@dataclass
+class ExtractionResult:
+    """Represents an extraction result."""
+
+    options: list[ExtractionOption]
+    """List of options from most confident to least confident."""
+
+    # explanation: str
+    # """Explanation of the given options. Empty if not specified."""
+
+
+def extract_from_recognized_texts(
+    recognized_texts: list[RecognizedText],
+) -> Optional[ExtractionResult]:
+    """Attempts to extract a result from the recognized texts by utilizing
+    ChatGPT.
+
+    Args:
+        recognized_texts: List of `RecognizedText` objects to extract from.
+    Returns:
+        An extraction result, which contains a list of options (pairs of
+        titles and authors).
+    """
+
+    # Create messages list to send as input
+    input_message = "\n".join(
```

Comment:
An example input in the form of a comment would be helpful in visualizing how the code sends in data to the GPT model